lib/resty/core/phase.lua

```diff
@@ -0,0 +1,57 @@
+local ffi = require 'ffi'
+local base = require "resty.core.base"
+
+local C = ffi.C
+local FFI_ERROR = base.FFI_ERROR
+local error = error
+local tostring = tostring
+
+
+ffi.cdef[[
+int ngx_http_lua_ffi_get_phase(ngx_http_request_t *r, char **err)
+]]
+
+
+local errmsg = base.get_errmsg_ptr()
+local CONTEXT_LOOKUP = {
```

Comment:
I think `context_names` is a better variable name here. We only use all-upper-case form for scalar constant variable names.